test/starknet/MerkleProof.test.ts

```diff
@@ -0,0 +1,89 @@
+import { computeHashOnElements } from 'starknet/dist/utils/hash';
+import { MerkleTree } from '../shared/merkle';
+
+describe('Merkle:', () => {
+  let testMerkle: StarknetContract;
+
+  before(async function () {
+    this.timeout(800000);
+    const testMerkleFactory = await starknet.getContractFactory(
+      './contracts/starknet/TestContracts/Test_Merkle.cairo'
+    );
+    testMerkle = await testMerkleFactory.deploy();
+  });
+
+  it('The library should verify a merkle proof for a given root', async () => {
+    const values = [];
+    // Generating random data for the merkle tree
+    for (let i = 0; i < 100; i++) {
+      values.push([
+        ethers.Wallet.createRandom().address,
+        ethers.utils.hexlify(ethers.utils.randomBytes(1)),
+      ]);
+    }
+
+    // computing the hash of each address value pair, and sorting
+    const merkleData = values
+      .map((v, i) => [computeHashOnElements(v), v[0], v[1]])
+      .sort(function (a, b) {
+        if (a > b) return 1;
+        if (a < b) return -1;
+        return 0;
+      })
+      .map((x, i) => [x[0], x[1], x[2], i]);
+    const leaves = merkleData.map((x) => x[0].toString());
+    const tree = new MerkleTree(leaves);
+
+    // Picking random leaf to prove
+    const address = values[Math.floor(Math.random() * 99)][0];
+    const leafData = merkleData.find((leaf) => leaf[1] == address)!;
+
+    await testMerkle.call('testAssertValidLeaf', {
+      root: tree.root,
+      leaf: [leafData[1], leafData[2]],
+      proof: tree.getProof(leaves, Number(leafData[3])),
+    });
+  }).timeout(600000);
+
+  it('The library should fail to verify if an invalid proof is supplied', async () => {
+    const values = [];
+    // Generating random data for the merkle tree
+    for (let i = 0; i < 100; i++) {
+      values.push([
+        ethers.Wallet.createRandom().address,
+        ethers.utils.hexlify(ethers.utils.randomBytes(1)),
+      ]);
+    }
+
+    // computing the hash of each address value pair, and sorting
+    const merkleData = values
+      .map((v, i) => [computeHashOnElements(v), v[0], v[1]])
+      .sort(function (a, b) {
+        if (a > b) return 1;
+        if (a < b) return -1;
+        return 0;
+      })
+      .map((x, i) => [x[0], x[1], x[2], i]);
+    const leaves = merkleData.map((x) => x[0].toString());
+    const tree = new MerkleTree(leaves);
+
+    // Picking random leaf to prove
+    const address = values[Math.floor(Math.random() * 99)][0];
+    const leafData = merkleData.find((leaf) => leaf[1] == address)!;
+
+    const corruptedProof = tree.getProof(leaves, Number(leafData[3]));
+    corruptedProof[0] = ethers.utils.hexlify(ethers.utils.randomBytes(4));
+    try {
+      await testMerkle.call('testAssertValidLeaf', {
+        root: tree.root,
+        leaf: [leafData[1], leafData[2]],
+        proof: corruptedProof,
+      });
+    } catch (error: any) {
+      expect(error.message).to.contain('Merkle: Invalid proof');
+    }
+  }).timeout(600000);
+});
```

Comment:
I'd add a couple of different tests:
- Empty merkle tree
- Merkle tree with 1 element
- With 2 elements
- With 3 elements
- With 10000 elements (a big one)
- Asking to prove a leaf that is NOT in the tree

Reply:
I added a test for empty, 1,2,3,1000 

tests for Non-membership proofs are not required (and are more sophisticated) so unnecessary to add here I think